### PR TITLE
Merge/0.5.dev2 code fixes to remove warnings

### PR DESCRIPTION
**Pull request by cbellot000**

### CODE REVIEW

File: ansys/dpf/core/data_tree.py [hunk 1]
```diff
@@ -25,7 +25,8 @@
 
 
 class DataTree:
-    """Represents an entity mapping attributes names to values.
+    """
```
Comment (PProfizi) [inline ansys/dpf/core/data_tree.py:28]: @cbellot000 I think there is a discrepancy between what flake8 or sphinx reports and what codacy reports regarding multiline docstrings beginning with an empy line. I think this version of writing it triggers a warning in one of the three.

File: ansys/dpf/core/dpf_array.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 """
+
```
Comment (PProfizi) [inline ansys/dpf/core/dpf_array.py:2]: Again, in my experience this triggers warnings in one of our other tools than the one you are currently looking at.

**Review by PProfizi — APPROVED** [review]

File: ansys/dpf/core/data_tree.py (hunk 1 above)
Comment (cbellot000) [reply ansys/dpf/core/data_tree.py:28]: @PProfizi I fixed it for codacy

File: ansys/dpf/core/server_types.py [hunk 3]
```diff
@@ -639,10 +639,8 @@ def shutdown(self):
         pass
 
     def __eq__(self, other_server):
-        """Return true, if ***** are equals"""
-        if isinstance(other_server, InProcessServer):
-            raise NotImplementedError
-        return False
+        """Return true, if the ip and the port are equals"""
+        return isinstance(other_server, InProcessServer)
```
[the remark this answers is not in the document]
Comment (cbellot000) [reply ansys/dpf/core/server_types.py:643]: @PProfizi if we are in process, there can be only one server

File: ansys/dpf/core/misc.py [hunk 4]
```diff
@@ -37,59 +24,6 @@ def module_exists(module_name):
     return module_name in (name for loader, name, ispkg in iter_modules())
 
 
-class Report(ScoobyReport):
```
[the remark this answers is not in the document]
Comment (cbellot000) [reply ansys/dpf/core/misc.py:40]: @PProfizi scooby was holding a reference on some of our object (in my case the issue was with a global variable named CLIENT_CONFIGURATION which is a DataTree, the reference on it prevented it from being properly destructed)

[the remark this answers is not in the document]
Comment (cbellot000) [reply ansys/dpf/core/misc.py:40]: @PProfizi do you think this report was used anywhere?

File: ansys/dpf/core/data_tree.py (hunk 1 above)
Comment (cbellot000) [reply ansys/dpf/core/data_tree.py]: reverted now

File: ansys/dpf/core/misc.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (PProfizi) [reply ansys/dpf/core/misc.py:40]: @cbellot000 oh okay, then I guess it is okay. We needed to change to the pyansys report tool anyway.

**Pull request merged**